server/server.go

```diff
@@ -378,7 +421,7 @@ func (s *Server) Publish(topic string, payload []byte, retain bool) error {
 	}
 
 	if retain {
-		s.retainMessage(pk)
+		s.retainMessage(&s.inline, pk)
```

Comment:
I wonder if this solves https://github.com/mochi-co/mqtt/issues/37 🤔

Reply:
Unfortunately this PR didn't fix #37, it seems like the retain flag is missing in the FixedHeader. I've created PR #42 which seems to fix the retain in my case.